// protoc -I=./message --cpp_out=./src ./MyMessage.proto
// protoc -I=$SRC_DIR --cpp_out=$DST_DIR $SRC_DIR/addressbook.proto
// 用户
message TCPUser
{
required string hostIP = 1;     // IP地址
required string hostPort = 2;   // 端口号
optional int32 socketid = 3;    // sockitID
optional string nick = 4;       // 昵称
}

message TCPMsg
{
required TCPUser fromUser = 1;

// 类型
enum TCPMsgType
{
EMsg_Msg = 1;       // 消息
EMsg_Cmd = 2;       // 命令
}

required TCPMsgType msgtype = 2;        // 类型

optional TCPMsgContent msgContent = 3;  // 消息部分
optional TCPCmdContent cmdContent = 4;  // 命令部分

}


message TCPMsgContent
{
required TCPUser toUser = 1;
required string  msgContent = 2;
}


message TCPCmdContent
{
required TCPUser toUser = 1;

enum TCPMsgType
{
ECmd_Login = 1;     // 登录
ECmd_Sync = 2;     // 同步列表
ECmd_Exit = 3;      // 退出
}

required TCPMsgType cmdType = 2;
optional string  cmdParam = 3;
repeated TCPUser  userList = 4;
}











